activejob/test/cases/exceptions_test.rb

```diff
@@ -191,19 +191,54 @@ class ExceptionsTest < ActiveSupport::TestCase
     ActiveJob::Base.retry_jitter = old_jitter
   end
 
+  test "random wait time for exponentially long wait and negative jitter value" do
+    old_jitter = ActiveJob::Base.retry_jitter
+    ActiveJob::Base.retry_jitter = -1.4
```

Comment:
I think we should be checking for a valid jitter when assigning a value to `ActiveJob::Base.retry_jitter`. If a user never passes a `jitter` value to `retry_on`, then getting an `ArgumentError` at runtime would be surprising to me and much more difficult to debug.

Reply:
@skipkayhil If the user does not pass `jitter` value, the error would not happen because the default value will kick in. 
This change, as I mentioned in the description, is more for dynamic values being passed which result in invalid values. 

Also, this flow will anyways cater to both the flows if the user somehow manages to mischievously add an invalid value to `ActiveJob::Base.retry_jitter` because that'll mostly be added statically.

Reply:
I think @skipkayhil's point was that putting the check in `determine_jitter_for_delay` will raise only when a job is retried.  Such an exception is easy to miss, and harder to track down.  It would be better to move the check to boot / configuration time.

We could rename `valid_jitter?` to `validate_jitter`, and put the `raise ArgumentError` inside of it, and then call it from `retry_on` and an override of `retry_jitter=`.  But that seems like a lot of defensive programming for an unlikely mistake.  @aditya-vector, is this an issue you've actually encountered in a real-world app?